pubpub/platform
# feat: shadcn sidebar

## Issue(s) Resolved

## High-level Explanation of PR


Move to new `@shadcn` sidebar component.

### Main changes

#### New sidebar

- Easier to maintain
- Better mobile experience


#### Move `base` color from `slate` to `gray`, so from a bluish gray to a more neutral one

this is bc our header and eg form config sidebar already have bg `gray-50`, so it makes sense to make the sidebar also `gray-50`. However, our default accent color `slate-100` does not look great on the `gray-50` bg, therefore @ktyjscott and I decided to move our main gray color to `gray` over `slate`. 
These are most of the changes in `ui`, and can be ignored.

#### Authorization checks

I changed the way authorization checks are done on the main page. Each link now has its own check, many of which are the same and cached.

Ideally, all these checks should be the same as the checks on the page itself, ideally-ideally we would define these checks in a separate place and use the same one for the page and the link to the page. ~I have not yet implemented this~. I have started working on this here, would love your thoughts https://github.com/pubpub/platform/pull/908


## Test Plan

1. Click on things, see if you can still get everywhere.
2. Log in as an editor (some@pubpub.org), check to see that you can't see that many things. (contributor will still fail)



## Screenshots (if applicable)

Demo

https://github.com/user-attachments/assets/635f6d96-9a0e-4752-9a61-99ef4e4fb988




## Notes
